android-sdk/src/main/java/ai/evolv/android_sdk/EvolvPredicatesImpl.java

```diff
@@ -134,13 +134,37 @@ private boolean evaluateFilter(JsonElement user, JsonObject rule) {
          //Evaluates a single filter rule against a user.
         //return !!FILTER_OPERATORS[rule.operator](value, rule.value);
         return false;
-
     }
 
     private JsonElement valueFromKey(JsonElement user, JsonObject rule) {
         // TODO: 10.06.2021
         return null;
     }
 
+    @FunctionalInterface
+    interface Function<A, B> {
+        boolean apply(A one, B two);
+    }
 
+    private Map<String, Function> operators = createOperatorsMap();
+
+    private Map<String, Function> createOperatorsMap() {
+        Map<String, Function> operatorsMap = new HashMap<>();
+
+        operatorsMap.put("contains", (Function<String, String>) (a, b) -> a.indexOf(b) >= 0);
+        // TODO: 11.06.2021  "defined" is the same as an "exists" (Android don't have undefined)!!!
+        operatorsMap.put("defined", (Function<String, String>) (a, b) ->  a !=null);
+        operatorsMap.put("equal", (Function<String, String>) (a, b) -> a == b);
+        operatorsMap.put("exists", (Function<String, String>) (a, b) -> a !=null);
+        operatorsMap.put("not_contains", (Function<String, String>) (a, b) -> !(a.indexOf(b) >= 0));
+        operatorsMap.put("not_defined", (Function<String, String>) (a, b) -> a !=null);
+        operatorsMap.put("not_equal", (Function<String, String>) (a, b) -> a != b);
+        operatorsMap.put("not_regex_match", (Function<String, String>) (a, b) -> false);
```

Comment:
Please make sure you return to the unimplemented regex operators

Reply:
I have several "todo" to implement some operators (**regex_match, regex64_match**)
and I should know about the "undefined" type (I will ask this question in slack)

Reply:
Answered in slack